go/vt/vtgate/plugin_mysql_server.go

```diff
@@ -155,6 +155,10 @@ func (vh *vtgateHandler) ComQuery(c *mysql.Conn, query string, callback func(*sq
 	}
 	session, result, err := vh.vtg.Execute(ctx, session, query, make(map[string]*querypb.BindVariable))
 	c.ClientData = session
+	if session != nil {
+		// Make sure the outcomeo of any potential USE statements are saved into the connection state
+		c.SchemaName = session.TargetString
```

Comment:
This shouldn't be needed any more (and to me seems incorrect).

Reply:
Oops yeah I thought I had reverted all of these. Will revert now.

Reply:
Ah I had just forgotten to push :-/